jujugui/static/gui/src/app/app.js

```diff
@@ -701,20 +689,38 @@ YUI.add('juju-gui', function(Y) {
       this.user.controller = { user, password, macaroons, external };
 
       controllerAPI.after('login', evt => {
+        const state = this.state;
+        const current = this.state.current;
         this.anonymousMode = false;
         if (evt.err) {
           this._renderLogin(evt.err);
           return;
         }
         this._renderUserMenu();
         console.log('successfully logged into controller');
+        // If state has a `next` property then that overrides all defaults.
+        const specialState = current.special;
+        const next = specialState && specialState.next;
+        // There should never be a `next` state if we aren't on login but
+        // adding the state login check here just to be sure.
+        if (state.current.root === 'login' && next) {
+          console.log('redirecting to "next" state', next);
+          const {error, state: newState} = state.generateState(next, false);
+          if (error === null) {
+            newState.root = null;
```

Comment:
Right, so this is because at this point `root` is going to be "login". If it's not set to `null` then the url will still have the `/login` until the model connects. This was handled by fluke elsewhere in the app :) I'll add a comment.